crates/router/src/connector/shift4/transformers.rs

```diff
@@ -602,3 +615,30 @@ pub struct ApiErrorResponse {
     pub code: Option<String>,
     pub message: String,
 }
+
+pub fn is_transaction_event(event: &Shift4WebhookEvent) -> bool {
+    matches!(
+        event,
+        Shift4WebhookEvent::ChargeCaptured
+            | Shift4WebhookEvent::ChargeFailed
+            | Shift4WebhookEvent::ChargeSucceeded
+            | Shift4WebhookEvent::ChargeUpdated
+    )
+}
+
+pub fn is_refund_event(event: &Shift4WebhookEvent) -> bool {
+    matches!(event, Shift4WebhookEvent::ChargeRefunded)
+}
+
+impl From<Shift4WebhookEvent> for api::IncomingWebhookEvent {
+    fn from(event: Shift4WebhookEvent) -> Self {
+        match event {
+            Shift4WebhookEvent::ChargeSucceeded | Shift4WebhookEvent::ChargeUpdated => {
```

Comment:
why ChargeSucceeded is also mapped as `PaymentIntentProcessing`
do we get `ChargeCaptured` for `automatic` capture_method also from shift4?

Reply:
[](https://dev.shift4.com/docs/api#event-types ) ChargeSucceded is when charge is created sucessfully , and charge captured Occurs when a charge is captured. No we are not getting chargeCaptured for automatic but , as per their documentation i have mapped the status .